From f2467a6042d3defbf139e58d10ef52b299e15d1d Mon Sep 17 00:00:00 2001
From: Dinesh Karthikeyan <quic_dinek@quicinc.com>
Date: Tue, 16 May 2023 11:58:48 +0530
Subject: [PATCH] ath12k: Invoke BUG_ON during bootup FW crash

When Q6 crash occurs while the board is in coldboot calibration, the
coldboot calibration gets timed out and dev flags for the radio is not
set to schedule reset work. Invoked BUG_ON to reboot the device.

Signed-off-by: Dinesh Karthikeyan <quic_dinek@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mhi.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

Index: backports-20220822-5.4.213-ef7197996efe/drivers/net/wireless/ath/ath12k/mhi.c
===================================================================
--- backports-20220822-5.4.213-ef7197996efe.orig/drivers/net/wireless/ath/ath12k/mhi.c	2023-05-23 12:13:12.886830583 +0530
+++ backports-20220822-5.4.213-ef7197996efe/drivers/net/wireless/ath/ath12k/mhi.c	2023-05-23 12:15:57.575287939 +0530
@@ -334,6 +334,9 @@ static void ath12k_mhi_op_status_cb(stru
 			set_bit(ATH12K_FLAG_RECOVERY, &ab->dev_flags);
 			queue_work(ab->workqueue_aux, &ab->reset_work);
 		}
+		else {
+			BUG_ON(1);
+		}
 		ath12k_hal_dump_srng_stats(ab);
 		break;
 	default:
